### PR TITLE
sci-physics/geant: Version bump to 4.10.03, EAPI bump 5 -> 6

### PR DESCRIPTION
**Pull request by g-braeunlich**

### CODE REVIEW

**Comment by g-braeunlich** [comment]
The files in app-doc/geant-docs did not change and are just renamed to match the current version.
Should I still create a new version for it or require the old version for geant-4.10.03?
Also version naming is different for geant and geant-docs. Should that be changed?

File: app-doc/geant-docs/geant-docs-4.10.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+SLOT="4"
+KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux"
+IUSE=""
+
+S="${DISTDIR}"
+
+src_install() {
+	local d
+	for d in *pdf; do newdoc ${d} ${d/-${PV}}; done
+	echo GEANT_DOC_DIR="${EPREFIX%/}/usr/share/doc/${PF}" >> 99geant-doc
```
Comment (SoapGentoo) [inline app-doc/geant-docs/geant-docs-4.10.3.ebuild]: missing `|| die`
also, `EPREFIX` doesn't need `%/`, it was never slash-terminated

**Comment by SoapGentoo** [comment]
@g-braeunlich do the versioning as you see fit

**Comment by SoapGentoo** [comment]
@g-braeunlich there's still something amiss, have you used `repoman manifest` everywhere?

**Comment by g-braeunlich** [comment]
I allways thougt that this is automatically invoked by repoman -dx commit.
Anyway, I will check.

File: sci-physics/geant-data/geant-data-4.10.03.ebuild [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+S="${WORKDIR}"
+
+src_unpack() {
+	# unpack in destination only to avoid copy
+	return
+}
+
+src_install() {
+	sed -n "s,export \(G4.\+DATA=\"\).*\(/share/Geant.\+/data/.\+\) > /dev/null ; pwd\`,\1${EROOT%/}/usr\2,p" \
+		"${EROOT}/usr/bin/geant4.sh" > 99geant-data
```
Comment (SoapGentoo) [inline sci-physics/geant-data/geant-data-4.10.03.ebuild]: `EROOT` makes no sense in `src_*` functions (they are purely related to cross-compiling), replace this by `EPREFIX`
Also, missing `|| die`

File: sci-physics/geant-data/geant-data-4.10.03.ebuild [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	doenvd 99geant-data
+	local g4dir=/usr/$(sed -n 's|.*/\(share/Geant4.*/data\).*|\1|p' "${EROOT}/usr/bin/geant4.sh" | tail -n 1)
```
Comment (SoapGentoo) [inline sci-physics/geant-data/geant-data-4.10.03.ebuild]: same here

File: sci-physics/geant-data/geant-data-4.10.03.ebuild [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	dodir ${g4dir}
+	cd "${ED%/}/${g4dir}"
```
Comment (SoapGentoo) [inline sci-physics/geant-data/geant-data-4.10.03.ebuild]: missing `|| die`

File: sci-physics/geant-data/geant-data-4.10.03.ebuild [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+	G4ABLA.3.0
+	G4PII.1.3
+	RealSurface.1.0
+	G4ENSDFSTATE.2.1
+	G4TENDL.1.3"
+
+SRC_COM="http://geant4.cern.ch/support/source"
+for d in ${GEANT4_DATA}; do
+	SRC_URI="${SRC_URI} ${SRC_COM}/${d}.tar.gz"
+done
```
Comment (SoapGentoo) [inline sci-physics/geant-data/geant-data-4.10.03.ebuild]: `unset d` after the loop (keep environment clean)

File: sci-physics/geant/geant-4.10.03.ebuild [hunk 6]
```diff
@@ -0,0 +1,85 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+inherit cmake-utils versionator multilib
```
Comment (SoapGentoo) [inline sci-physics/geant/geant-4.10.03.ebuild]: do you still need `multilib`?

File: sci-physics/geant/geant-4.10.03.ebuild [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+RDEPEND="
+	dev-libs/expat
+	>=sci-physics/clhep-2.3.3.0:2=
+	dawn? ( media-gfx/dawn )
+	gdml? ( dev-libs/xerces-c )
+	motif? ( x11-libs/motif:0 )
+	opengl? ( virtual/opengl )
+	inventor? ( media-libs/SoXt )
+	qt4? ( dev-qt/qtgui:4 opengl? ( dev-qt/qtopengl:4 ) )
```
Comment (SoapGentoo) [inline sci-physics/geant/geant-4.10.03.ebuild]: put these on two lines

File: sci-physics/geant/geant-4.10.03.ebuild [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+	raytracerx? ( x11-libs/libX11 x11-libs/libXmu )
```
Comment (SoapGentoo) [inline sci-physics/geant/geant-4.10.03.ebuild]: same here (one dep per line)

File: sci-physics/geant/geant-4.10.03.ebuild [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+	zlib? ( sys-libs/zlib )"
+DEPEND="${RDEPEND} prefix? ( app-shells/tcsh )"
+PDEPEND="
+	data? ( ~sci-physics/geant-data-${PV} )
+	doc? ( ~app-doc/geant-docs-${PV} )"
+
+DOCS=( ReleaseNotes/ReleaseNotes${SPV}.html )
```
Comment (SoapGentoo) [inline sci-physics/geant/geant-4.10.03.ebuild]: maybe `HTML_DOCS` instead (your call)?

File: sci-physics/geant/geant-4.10.03.ebuild [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+
+S="${WORKDIR}/${MY_P}"
+
+src_prepare() {
+	use examples || sed -i '/install(DIRECTORY examples/,/)/d' CMakeLists.txt || die
```
Comment (SoapGentoo) [inline sci-physics/geant/geant-4.10.03.ebuild]: you'll have to turn this into a proper `if; then` for the `|| die` to work

File: sci-physics/geant/geant-4.10.03.ebuild [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+	default
```
Comment (SoapGentoo) [inline sci-physics/geant/geant-4.10.03.ebuild]: cmake-utils -> never use `default`, but `cmake-utils_src_prepare` (and call it first in `src_prepare`)

**Comment by g-braeunlich** [comment]
You were right, the manifest for geant-docs has not been updated. Fixed now.

File: sci-physics/geant/geant-4.10.03.ebuild (hunk 10 above)
Comment (g-braeunlich) [inline sci-physics/geant/geant-4.10.03.ebuild]: Btw: why is
A || B || die
not working?

Comment (SoapGentoo) [inline sci-physics/geant/geant-4.10.03.ebuild]: See:
https://devmanual.gentoo.org/ebuild-writing/error-handling/index.html
those constructs don't work as you expect them too

**Pull request closed without merging**